src/base.jl

```diff
@@ -661,6 +661,14 @@ function remove_component!(
     handle_component_removal!(sys, component)
 end
 
+"""
+Check to see if the component of type T with name exists. If T is an abstract
+type then the names of components across all subtypes of T must be unique.
```

Comment:
You can delete `If T is an abstract type then the names of components across all subtypes of T must be unique.`, right? We do allow duplicate names across subtypes of an abstract type. There is a limitation is in `get_component`. That function will throw an exception if you specify an abstract type with a name that is duplicate. It will work if you specify the concrete type.

This is something that will likely confuse users.